include/nbl/asset/filters/kernels/CConvolutionWeightFunction.h

```diff
@@ -0,0 +1,175 @@
+			}
+
+			double result = 0.0;
+			for (uint32_t i = 0u; i < sampleCount; ++i)
+			{
+				const double tau = minIntegrationLimit + i * dtau;
+				if (m_isFuncAWider)
+					result += m_funcA.weight(tau, channel) * m_funcB.weight(x - tau, channel) * dtau;
+				else
+					result += m_funcB.weight(tau, channel) * m_funcA.weight(x - tau, channel) * dtau;
+			}
+			return static_cast<float>(result);
+		}
+	}
+
+	std::pair<double, double> getIntegrationDomain(const float x) const
+	{
+		// We assume that the wider function is stationary (not shifting as `x` changes) while the narrower function is the one which shifts, such that it is always centered at x.
+
+		const float funcNarrowMinSupport = m_isFuncAWider ? m_funcB.getMinSupport() : m_funcA.getMinSupport();
+		const float funcNarrowMaxSupport = m_isFuncAWider ? m_funcB.getMaxSupport() : m_funcA.getMaxSupport();
+
+		const float funcWideMinSupport = m_isFuncAWider ? m_funcA.getMinSupport() : m_funcB.getMinSupport();
+		const float funcWideMaxSupport = m_isFuncAWider ? m_funcA.getMaxSupport() : m_funcB.getMaxSupport();
+
+		const float funcNarrowWidth = funcNarrowMaxSupport - funcNarrowMinSupport;
+		const float funcWideWidth = funcWideMaxSupport - funcWideMinSupport;
+
+		const float funcNarrowWidth_half = funcNarrowWidth * 0.5;
+
+		double minIntegrationLimit = 0.0, maxIntegrationLimit = 0.0;
+		{
+			if ((x >= (funcWideMinSupport - funcNarrowWidth_half)) && (x <= (funcWideMinSupport + funcNarrowWidth_half)))
+			{
+				minIntegrationLimit = funcWideMinSupport;
+				maxIntegrationLimit = x + funcNarrowWidth_half;
+			}
+			else if ((x >= (funcWideMinSupport + funcNarrowWidth_half)) && (x <= (funcWideMaxSupport - funcNarrowWidth_half)))
+			{
+				minIntegrationLimit = x - funcNarrowWidth_half;
+				maxIntegrationLimit = x + funcNarrowWidth_half;
+			}
+			else if ((x >= (funcWideMaxSupport - funcNarrowWidth_half)) && (x <= (funcWideMaxSupport + funcNarrowWidth_half)))
+			{
+				minIntegrationLimit = x - funcNarrowWidth_half;
+				maxIntegrationLimit = funcWideMaxSupport;
+			}
+		}
+		assert(minIntegrationLimit <= maxIntegrationLimit);
+
+		return { minIntegrationLimit, maxIntegrationLimit };
+	}
+};
+
+template <>
+double CConvolutionWeightFunction1D<CWeightFunction1D<SBoxFunction>, CWeightFunction1D<SBoxFunction>>::weight_impl(const float x, const uint32_t channel, const uint32_t) const;
+
+template <>
+double CConvolutionWeightFunction1D<CWeightFunction1D<SGaussianFunction>, CWeightFunction1D<SGaussianFunction>>::weight_impl(const float x, const uint32_t channel, const uint32_t) const;
+
+// TODO: Specialization: CConvolutionWeightFunction1D<CWeightFunction1D<STriangleFunction>, CWeightFunction1D<STriangleFunction>> = this is tricky but feasible
+
+template <>
+double CConvolutionWeightFunction1D<CWeightFunction1D<SKaiserFunction>, CWeightFunction1D<SKaiserFunction>>::weight_impl(const float x, const uint32_t channel, const uint32_t) const;
```

Comment:
For Box it would be zero everywhere except where it is not defined.

For Kaiser it doesn't seem very straight forward to integrate the first order derivative of the bessel function thingy.

Not sure about Gaussian, but seems doable at least.

Reply:
Leave it for later/when needed I guess.